pkg/sql/copy_file_upload.go

```diff
@@ -108,13 +118,16 @@ func CopyInFileStmt(destination, schema, table string) string {
 	)
 }
 
-func (f *fileUploadMachine) run(ctx context.Context) error {
-	err := f.c.run(ctx)
-	_ = f.writeToFile.Close()
-	if err != nil {
-		return err
-	}
-	return <-f.writeError
+func (f *fileUploadMachine) run(ctx context.Context) (err error) {
+	err = f.c.run(ctx)
+	defer func() {
```

Comment:
this defer doesn't do anything extra. you can just put the logic right in the function.